### ISSUE TITLE
cfn-lint should validate json properly

### ISSUE THREAD

**Issue by paololazzari** [aws-cloudformation/cfn-lint#2655]
### Describe the feature you'd like to request

```bash
$ cfn-lint -v
cfn-lint 0.76.1
```

Consider template `template.json`:

```json
{
    "Resources": {
        "MyEC2Instance": {
            "Type": "AWS::EC2::Instance",
            "Properties": {
                "ImageId": "ami-79fd7eee",
                "KeyName": "testkey",
                "BlockDeviceMappings": [
                    {
                        "DeviceName": "/dev/sdm",
                        "Ebs": {
                            "VolumeType": "io1",
                            "Iops": "200",
                            "DeleteOnTermination": "false",
                            "VolumeSize": "20"
                        }
                    },
                    {
                        "DeviceName": "/dev/sdk",
                        "NoDevice": {}
                    }
                ]
            }
        }
    },
}
```

this isn't valid json:

```bash
$ jq '.' template.json
parse error: Expected another key-value pair at line 21, column 9
```

yet `cfn-lint` seems to think it is (or doesn't care)

```bash
$ cfn-lint template.json
$ echo $?
0
```

### Describe the solution you'd like

`cfn-lint` should validate json properly 

### Is this something that you'd be interested in working on?

- [ ✓ ] 👋 I may be able to implement this feature request

### Would this feature include a breaking change?

- [ X ] ⚠️ This feature might incur a breaking change

**Comment by kddejong** [comment]
Confirmed... I think this is because it hits the yaml parser first but I would have to double check that.

**Comment by kddejong** [comment]
Oddly enough is also okay in the cloudformation service

**Comment by kftsehk** [comment]
I think the only "wrong" of this json is here

```json
{
    "Resources": {
        "MyEC2Instance": {
            "Type": "AWS::EC2::Instance",
            "Properties": {
                "ImageId": "ami-79fd7eee",
                "KeyName": "testkey",
                "BlockDeviceMappings": [
                    {
                        "DeviceName": "/dev/sdm",
                        "Ebs": {
                            "VolumeType": "io1",
                            "Iops": "200",
                            "DeleteOnTermination": "false",
                            "VolumeSize": "20"
                        }
                    },
                    {
                        "DeviceName": "/dev/sdk",
                        "NoDevice": {}
                    }
                ]
            }
        }
   // here extra comma
    },
}
```

It is not a strictly standard json.
But the question is, if this json consider a valid representation of the resource you are trying to deploy, is likely up to the parser implementation. Some parser can be more permissive and attempt to work with that comma removed/ignored.

![image](https://user-images.githubusercontent.com/20808792/230470532-2a820240-e209-4690-aca0-19f4eaaa2bc2.png)

E.g. I've fixed the json simply passing it through a popular JS/JSON formatter - Prettier.

![image](https://user-images.githubusercontent.com/20808792/230470645-3e3a8605-caf2-4469-85bb-e38066647eb7.png)

**Comment by kddejong** [comment]
I think I'm going to close this one since this template is okay with the CloudFormation service which is what we are trying to validate against. If you want to validate its valid json you may need to use a different solution.
